test/test_torch.py

```diff
@@ -12671,16 +12671,25 @@ def test_topk_noncontiguous_gpu(self, device):
         self.assertEqual(top1, top2)
         self.assertEqual(idx1, idx2)
 
+    @dtypes(torch.int8, torch.uint8, torch.int16, torch.int32, torch.int64)
+    def test_topk_integral(self, device, dtype):
+        a = torch.randint(torch.iinfo(dtype).min, torch.iinfo(dtype).max, size=(10,),
+                          dtype=dtype, device=device)
+        sort_topk = a.sort()[0][-5:].flip(0)
+        topk = a.topk(5)
+        self.assertEqual(sort_topk, topk[0])      # check values
+        self.assertEqual(sort_topk, a[topk[1]])   # check indices
+
     def test_topk_nonfinite(self, device):
-        for dtype in (torch.float, torch.double):
-            x = torch.tensor([float('nan'), float('inf'), 1e10, 0, -1e10, -float('inf')], device=device)
+        for dtype in (torch.half, torch.float, torch.double):
```

Comment:
please put dtypes in the decorator, not in the loop here.

Reply:
Changed to
```
    @dtypesIfCUDA(torch.half, torch.float, torch.double)
    @dtypes(torch.float, torch.double)
    def test_topk_nonfinite(self, device, dtype):
```
Asking @ailzhang in other PR if something else needs to be done to allow a bfloat16 case for XLA.  Possibly bfloat16 should be tested on CPU as well?

Reply:
Does chaining the two decorators act the way we want?  I copied that pattern from other tests.

Reply:
Hi @mcarilli, thanks for the change! 
Is it true that CUDA supports `torch.half` while CPU supports `torch.bfloat16` in topk?

Reply:
Cuda supports torch.half, or will again after this PR.

I'm not sure if CPU supports bfloat16, but my changes are in a Cuda-specific file, so they probably don't affect the CPU code path in any case.  If the CPU does support bfloat16 topk, adding `dtypes(..., torch.bfloat16)` may be a good idea regardless, but won't test this PR's code changes.

The bfloat16 changes in this PR are definitely necessary, but because Cuda doesn't support bfloat16 yet, I'm not sure how (or if) we can include a test for them.